### PR TITLE
New Syntax Change ( structs to objects/records )

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
fixed issue #7 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: jira7/Jira_connector_constants.bal [hunk 1]
```diff
@@ -22,55 +22,56 @@ package jira7;
...
 
-public const int STATUS_CODE_OK = 200;
-public const int STATUS_CODE_CREATED = 201;
-public const int STATUS_CODE_NO_CONTENT = 204;
-public const int STATUS_CODE_NOTFOUND = 404;
-public const int STATUS_CODE_INTERNAL_ERROR = 500;
-public const int STATUS_CODE_BAD_REQUEST = 400;
-public const int STATUS_CODE_UNAUTHORIZED = 401;
-public const int Status_CODE_CONFLICT = 409;
+public int STATUS_CODE_OK = 200;
```
Comment (muthulee) [inline jira7/Jira_connector_constants.bal]: Should we use @final public int?

File: jira7/jira_connector.bal [hunk 2]
```diff
@@ -24,36 +24,105 @@ import ballerina/log;
 import ballerina/io;
 
 //Jira Connector Struct
```
Comment (muthulee) [inline jira7/jira_connector.bal]: Please remove package statement from here.

File: jira7/tests/sample.bal [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import ballerina/io;
```
Comment (muthulee) [inline jira7/tests/sample.bal]: please remove this file.

File: jira7/jira_connector_endpoint.bal [hunk 4]
```diff
@@ -21,48 +21,45 @@ package jira7;
...
+public type JiraConfiguration {
+            http:ClientEndpointConfiguration httpClientConfig;
+string url;
+string username;
+string password;
+};
+
+
+
+public type JiraEndpoint object {
```
Comment (muthulee) [inline jira7/jira_connector_endpoint.bal]: Please rename Endpoint to client.

File: jira7/jira_connector_endpoint.bal [hunk 5]
```diff
@@ -21,48 +21,45 @@ package jira7;
...
+    public {
+        JiraConfiguration jiraConfig;
+        JiraConnector jiraConnector;
+    }
+
```
Comment (muthulee) [inline jira7/jira_connector_endpoint.bal]: Endpoint needs start, stop, register methods.

File: jira7/Jira_connector_constants.bal (hunk 1 above)
Comment (NipunaRanasinghe) [reply jira7/Jira_connector_constants.bal]: +1, defined as @final

Comment (NipunaRanasinghe) [reply jira7/Jira_connector_constants.bal]: +1, defed as @final

File: jira7/jira_connector.bal (hunk 2 above)
Comment (NipunaRanasinghe) [reply jira7/jira_connector.bal]: Removed package declaration

File: jira7/jira_connector_endpoint.bal (hunk 4 above)
Comment (NipunaRanasinghe) [reply jira7/jira_connector_endpoint.bal]: Renamed the endpoint object as "JiraClient"

File: jira7/jira_connector_endpoint.bal (hunk 5 above)
Comment (NipunaRanasinghe) [reply jira7/jira_connector_endpoint.bal]: Added the additional methods.

File: jira7/tests/sample.bal (hunk 3 above)
Comment (NipunaRanasinghe) [reply jira7/tests/sample.bal]: Removed the sample file

**Pull request merged**